README.md

```diff
@@ -43,3 +43,7 @@ Thank you:
 * And to everyone who has contributed to this repository or filed a bug report.
 
 My code is still a bit of a mess but utility over cleanliness, will continue to fix up in the coming days after the more critical work is over.
+
+## Tech notes
+
+See [this doc](/blob/master/tech-notes.md) for tech notes.        
```

Comment:
What are the expectations for keeping the tech notes up-to-date? e.g., should each pull request be required to also include any relevant changes to tech-notes.md?

Reply:
That's a really good idea - @everestpipkin could that be a condition of accepting a PR?

Reply:
Yes, I'd appreciate it if folks will document their own changes! We can request it in the tech notes maybe. :) That said, I can also try to add notes as things are changes/updated if I see they're missing.